Paintroid/src/main/java/org/catrobat/paintroid/tools/implementation/BaseToolWithRectangleShape.java

```diff
@@ -233,9 +239,15 @@ public boolean handleMove(PointF coordinate) {
 		return true;
 	}
 
-	protected void onClick(PointF coordinate) {
-		if (boxContainsPoint(coordinate)) {
-			onClickInBox();
+	protected void onClickButton() {
+		ImageButton button = checkmarkButton.findViewById(R.id.pocketpaint_btn_top_checkmark);
+		if (button != null) {
+			button.setOnClickListener(new View.OnClickListener() {
+				@Override
+				public void onClick(View view) {
+					onClickOnButton();
+				}
+			});
```

Comment:
in this case you would register this listener quite often. move it to `MainActivity::setTopBarListeners()` instead. you can set a listener on a view that is `View.GONE`.